controllers/templatesync/template_sync.go

```diff
@@ -1170,14 +1178,22 @@ func (r *PolicyReconciler) processDependencies(
 		}
 
 		depObj, err := res.Get(ctx, dep.Name, metav1.GetOptions{})
-		if err != nil {
+		if k8serrors.IsNotFound(err) {
```

Comment:
I think you'll also need to check `errors.As(err, &errGroupDiscoveryFailed)` since Gatekeeper/ConstraintTemplate CRD is not guaranteed to be installed.

Reply:
Actually, I see further up you error early if the GVR can't be found.

Reply:
I kinda thought GVRFromGVK would handle that kind of thing.